whatlies/embeddingset.py

```diff
@@ -203,9 +203,12 @@ def __rshift__(self, other):
         new_embeddings = {k: emb >> other for k, emb in self.embeddings.items()}
         return EmbeddingSet(new_embeddings, name=f"({self.name} >> {other.name})")
 
-    def compare_against(self, other, mapping="direct"):
+    def compare_against(self, other, mapping="direct", func=None):
         if mapping == "direct":
-            return [v > other for k, v in self.embeddings.items()]
+            if func is None:
+                return [v > other for v in self.embeddings.values()]
+            else:
+                return [func(v.vector, other.vector) for v in self.embeddings.values()]
```

Comment:
Might it be nicer to do something like; 

```
if callable(mapping):
    return [mapping(v.vector, other.vector) for v in self.embeddings.values()]
```

I think this was my original idea with that function as well.

Reply:
Nitpick; I suppose a unit test for this method can't hurt.

Reply:
Ah, right. I didn't think about that. Made some changes and tests to address this.